### PR TITLE
Clickable Icons: Clear cache search cache for get_details request based on a request flag

### PR DESCRIPTION
**Pull request by shubham1g5**
## Product Description
https://dimagi-dev.atlassian.net/browse/QA-5771

Fixes issue where get_details request doesn't give latest data after performing a case change using clickable icon due to cache search cache not refreshing.

## Technical Summary

Adds a flag called `is_refresh_case_search` to request beans and clear cache search cache if it's set to `true`

### Safety story
Should only affect clickable icons feature when we set the `isRefreshCaseSearch` to true for `get_details`

### Automated test coverage
PR adds a test to verify new bheaviour

### QA Plan
None

### Special deploy instructions

- [x] This PR can be deployed after merge with no further considerations.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations.

### Review

- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change.

### CODE REVIEW

File: src/main/java/org/commcare/formplayer/beans/SessionNavigationBean.java [hunk 1]
```diff
@@ -118,6 +120,9 @@ public void setIsPersistent(boolean persistent) {
     public boolean getIsShortDetail() {
         return isShortDetail;
     }
+    public boolean getIsRefreshCaseSearch() {
```
Comment (MartinRiese) [inline src/main/java/org/commcare/formplayer/beans/SessionNavigationBean.java:123]: I thought standard naming leaves the get off the name if the field is a boolean.

**Review by MartinRiese — APPROVED** [review]
Just a naming nit

File: src/main/java/org/commcare/formplayer/beans/SessionNavigationBean.java (hunk 1 above)
Comment (shubham1g5) [reply src/main/java/org/commcare/formplayer/beans/SessionNavigationBean.java:123]: yea, although the naming for response beans is a little different as  this gets used by the Json coverter to convert a json for this class and the standand naming convention there is `get_{$var}` and `set_{$var}`

Comment (Robert-Costello) [reply src/main/java/org/commcare/formplayer/beans/SessionNavigationBean.java:123]: Does this need a setter or does it function differently from `isShortDetail`?

Comment (shubham1g5) [reply src/main/java/org/commcare/formplayer/beans/SessionNavigationBean.java:123]: it does, corrected

**Pull request merged**